helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -647,9 +650,16 @@ private static PipelineRegistry createManagementModeRegistry(String pipelineName
       Pipeline dataRefresh = new Pipeline(pipelineName);
       dataRefresh.addStage(new ReadClusterDataStage());
 
+      // data pre-process pipeline
+      Pipeline dataPreprocess = new Pipeline(pipelineName);
+      dataPreprocess.addStage(new ResourceComputationStage());
```

Comment:
NIT: Do we need ResourceValidationStage? It seemed to bundle with ResourceComputationStage.

Reply:
No. ResourceValidation is to match the ideal state rules, which is not required management pipeline. We can assume in management pipeline, the ideal states are validated since management pipeline is switched from default pipeline.
The pipeline switch logic from default to management is ResourceValidation. To avoid handling it in management pipeline and make the logic simple, I don't want to add ResourceValidation.